codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocket08FrameEncoder.java

```diff
@@ -186,8 +185,10 @@ protected void encode(ChannelHandlerContext ctx, WebSocketFrame msg, List<Object
                     int end = data.writerIndex();
 
                     if (srcOrder == dstOrder) {
-                        // Use the optimized path only when byte orders match
-                        long longMask = mask | (long) mask << 32;
+                        // Use the optimized path only when byte orders match.
+                        // Avoid sign extension on widening primitive conversion
+                        long longMask = (long) mask & 0xFFFFFFFFL;
```

Comment:
This change is not strictly necessary because ThreadLocalRandom.nextInt(limit) returns non-negative values. However theoretically mask may be represented with negative int (if highest bit is set). Original logic before https://github.com/netty/netty/pull/12969 also assumed possibility of negative mask.